components/calendar/src/any_calendar.rs

```diff
@@ -0,0 +1,324 @@
+                // attempt to convert
+                let iso = calendar2.date_to_iso(&date2);
+
+                match_cal_and_date!(match (self, date1):
+                    (c1, d1) => {
+                        let d2 = c1.date_from_iso(iso);
+                        let until = c1.until(d1, &d2, &c1, largest_unit, smallest_unit);
+                        until.cast_unit::<AnyCalendar>()
+                    }
+                )
+            }
+        }
+    }
+
+    /// The calendar-specific year represented by `date`
+    fn year(&self, date: &Self::DateInner) -> types::Year {
+        match_cal_and_date!(match (self, date): (c, d) => c.year(d))
+    }
+
+    /// The calendar-specific month represented by `date`
+    fn month(&self, date: &Self::DateInner) -> types::Month {
+        match_cal_and_date!(match (self, date): (c, d) => c.month(d))
+    }
+
+    /// The calendar-specific day-of-month represented by `date`
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        match_cal_and_date!(match (self, date): (c, d) => c.day_of_month(d))
+    }
+
+    /// Information of the day of the year
+    fn day_of_year_info(&self, date: &Self::DateInner) -> types::DayOfYearInfo {
+        match_cal_and_date!(match (self, date): (c, d) => c.day_of_year_info(d))
+    }
+
+    fn debug_name(&self) -> &'static str {
+        match *self {
+            Self::Gregorian(_) => "AnyCalendar (Gregorian)",
+            Self::Buddhist(_) => "AnyCalendar (Buddhist)",
+            Self::Japanese(_) => "AnyCalendar (Japanese)",
+            Self::Indian(_) => "AnyCalendar (Indian)",
+            Self::Coptic(_) => "AnyCalendar (Coptic)",
+            Self::Iso(_) => "AnyCalendar (Iso)",
+        }
+    }
+}
+
+impl AnyCalendar {
+    fn calendar_name(&self) -> &'static str {
+        match *self {
+            Self::Gregorian(_) => "Gregorian",
+            Self::Buddhist(_) => "Buddhist",
+            Self::Japanese(_) => "Japanese",
+            Self::Indian(_) => "Indian",
+            Self::Coptic(_) => "Coptic",
+            Self::Iso(_) => "Iso",
+        }
+    }
+}
+
+impl AnyDateInner {
+    fn calendar_name(&self) -> &'static str {
+        match *self {
+            AnyDateInner::Gregorian(_) => "Gregorian",
+            AnyDateInner::Buddhist(_) => "Buddhist",
+            AnyDateInner::Japanese(_) => "Japanese",
+            AnyDateInner::Indian(_) => "Indian",
+            AnyDateInner::Coptic(_) => "Coptic",
+            AnyDateInner::Iso(_) => "Iso",
+        }
+    }
+}
+
+impl<C: IncludedInAnyCalendar> From<C> for AnyCalendar {
+    fn from(c: C) -> AnyCalendar {
+        c.to_any()
+    }
+}
+
+/// Trait for calendars that may be converted to [`AnyCalendar`]
+pub trait IncludedInAnyCalendar: Calendar + Sized {
+    /// Convert this calendar into an [`AnyCalendar`], moving it
+    ///
+    /// You should not need to call this method directly
+    fn to_any(self) -> AnyCalendar {
+        self.to_any_cloned()
+    }
```

Comment:
Nit: I'd prefer if this weren't defaulted; implementing it manually is not that hard and is less prone to error in calendars where this method is useful

Reply:
I don't think this comment is resolved; the default impl still exists. I wanted you to delete it.

Reply:
oh, looks like i only did this partially. done